src/auth/auth-api-request.ts

```diff
@@ -785,6 +851,62 @@ export abstract class AbstractAuthRequestHandler {
     return this.invokeRequestHandler(this.getAuthUrlBuilder(), FIREBASE_AUTH_GET_ACCOUNT_INFO, request);
   }
 
+  /**
+   * Looks up multiple users by their identifiers (uid, email, etc).
+   *
+   * @param {UserIdentifier[]} identifiers The identifiers indicating the users
+   *     to be looked up. Must have <= 100 entries.
+   * @param {Promise<object>} A promise that resolves with the set of successfully
+   *     looked up users. Possibly empty if no users were looked up.
+   */
+  public getAccountInfoByIdentifiers(identifiers: UserIdentifier[]): Promise<object> {
+    if (identifiers.length === 0) {
+      return Promise.resolve({users: []});
+    } else if (identifiers.length > 100) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.MAXIMUM_USER_COUNT_EXCEEDED,
+        '`identifiers` parameter must have <= 100 entries.');
+    }
+
+    const request: GetAccountInfoRequest = {};
+
+    for (const id of identifiers) {
+      if (isUidIdentifier(id)) {
```

Comment:
I feel like a polymorphic design would have cleaned this clutter up a lot, but such APIs are not very user friendly in JS. How about we at least push some of this logic into small helper functions?

```
if (isUserIdentifier(id)) {
  handleUserIdentifier(id, request);
} else if (isEmailIdentifier(id)) {
  handleEmailIdentifier(id, request);
}
// and so on...
```